README.md

```diff
@@ -36,13 +37,50 @@ Supports most Debian and RHEL-based Linux distributions, and Windows.
 |-------------------------------------------------------------------------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
 | `datadog_api_key`                                                                                                                               | Your Datadog API key.                                                                                                                                                                                                                                    |
 | `datadog_site`                                                                                                                                  | The site of the Datadog intake to send Agent data to. Defaults to `datadoghq.com`, set to `datadoghq.eu` to send data to the EU site. This option is only available with agent version >= 6.6.0.                                                         |
-| `datadog_agent_version`                                                                                                                         | The pinned version of the Agent to install (optional, but highly recommended). Examples: `1:6.0.0-1` on apt-based platforms, `6.0.0-1` on yum-based platforms, `6.0.0` on Windows platforms. **Note** Downgrades are not supported on Windows platforms. |
+| `datadog_agent_major_version`                                                                                                                         | The major version of the Agent which will be installed. Its possible values are 5, 6, 7. If both this and `datadog_agent_version` are not set, then `datadog_agent_major_version` defaults to 7. If this is not set but `datadog_agent_version` is set, then `datadog_agent_major_version` is set to the major version provided in `datadog_agent_version`. |
+| `datadog_agent_version`                                                                                                                         | The pinned version of the Agent to install (optional, but highly recommended). Examples: `1:6.0.0-1` on apt-based platforms, `6.0.0-1` on yum-based platforms, `6.0.0` on Windows platforms. If both this variable and `datadog_agent_major_version` are set, they must be compatible (the major version in `datadog_agent_version` has to be `datadog_agent_major_version`). **Note** Downgrades are not supported on Windows platforms. |
```

Comment:
```suggestion
| `datadog_agent_version`                                                                                                                         | The pinned version of the Agent to install (optional, but highly recommended). Example: `7.15.0`. Using `datadog_agent_major_version` is not needed is `datadog_agent_version` is used.. **Note** Downgrades are not supported on Windows platforms. |
```

I think we should first document `datadog_agent_version` then `datadog_agent_major_version` as we advice people to use `datadog_agent_version` over `datadog_agent_major_version`.

I don't think we need to document that both `datadog_agent_version` and `datadog_agent_major_version` should be compatible. The main idea is not to use both at the same time.